### PR TITLE
Changes the sort logic.

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**

### CODE REVIEW

File: src/utils/CommonUtils.js [hunk 1]
```diff
@@ -41,3 +43,32 @@ export const paginatedData = ({ currentData, recordsPerPage, currentPage }) => {
 
   return [];
 };
+
+export const getSortedData = ({ columnName, columnType, sortOrder, data }) => {
+  let temporaryData = cloneDeep(data);
+  if (columnType === 'Number') {
+    let stringContained = [];
+    let numberContained = [];
+    let undefinedContained = [];
```
Comment (pepcus-pratik) [inline src/utils/CommonUtils.js]:
```suggestion
    let emptyValues = [];
```

File: src/utils/CommonUtils.js [hunk 2]
```diff
@@ -41,3 +43,32 @@ export const paginatedData = ({ currentData, recordsPerPage, currentPage }) => {
...
+    temporaryData.forEach((object) => {
+      if (object[columnName] === "") {
+        undefinedContained.push(object);
```
Comment (pepcus-pratik) [inline src/utils/CommonUtils.js]: Improve the code logic

File: src/components/dataGrid/Table.jsx [hunk 3]
```diff
@@ -54,7 +54,13 @@ class Table extends Component {
     if (!isEqual(nextProps.data, this.props.data)
       || !isEqual(nextProps.metaData, this.props.metaData)) {
       let temporaryData = filterData(this.state.appliedFilter, nextProps.data);
-      temporaryData = orderBy(temporaryData, this.state.sort.columnName, this.state.sort.sortOrder);
+      temporaryData = getSortedData(
+        {
+          columnName: this.state.sort.columnName,
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/dataGrid/Table.jsx]: Done.

File: src/components/dataGrid/Table.jsx [hunk 4]
```diff
@@ -141,8 +149,13 @@ class Table extends Component {
       ...appliedFilter,
       [inputValue.selectedColumn]: inputValue.searchString,
     };
-    const temporaryData
-      = orderBy(this.state.originalData, this.state.sort.columnName, this.state.sort.sortOrder);
+    const temporaryData = getSortedData(
+      {
+        columnName: this.state.sort.columnName,
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/dataGrid/Table.jsx]: Done.

File: src/components/dataGrid/Table.jsx [hunk 5]
```diff
@@ -158,19 +171,31 @@ class Table extends Component {
...
 
-  onSort(columnId) {
+  onSort(columnId, columnType) {
     if (this.state.sort.sortOrder === 'asc') {
-      const sortedData = orderBy(this.state.currentData, [columnId], ['desc']);
+      const sortedData = getSortedData(
+        {
+          columnName: columnId,
+          columnType: columnType,
+          sortOrder: 'desc',
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/components/dataGrid/Table.jsx]: Done.

File: src/utils/CommonUtils.js [hunk 6]
```diff
@@ -41,3 +43,32 @@ export const paginatedData = ({ currentData, recordsPerPage, currentPage }) => {
 
   return [];
 };
+
+export const getSortedData = ({ columnName, columnType, sortOrder, data }) => {
+  let temporaryData = cloneDeep(data);
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/utils/CommonUtils.js]: Done.

File: src/utils/CommonUtils.js [hunk 7]
```diff
@@ -41,3 +43,32 @@ export const paginatedData = ({ currentData, recordsPerPage, currentPage }) => {
...
+  if (columnType === 'Number') {
+    let stringContained = [];
+    let numberContained = [];
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/utils/CommonUtils.js]: Done.

File: src/utils/CommonUtils.js (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/utils/CommonUtils.js]: Done.

File: src/utils/CommonUtils.js [hunk 8]
```diff
@@ -41,3 +43,32 @@ export const paginatedData = ({ currentData, recordsPerPage, currentPage }) => {
 
   return [];
 };
+
+export const getSortedData = ({ columnName, columnType, sortOrder, data }) => {
+  let temporaryData = cloneDeep(data);
+  if (columnType === 'Number') {
+    let stringContained = [];
```
[the remark this answers is not in the document]
Comment (BhagyashreeBhoyar) [reply src/utils/CommonUtils.js]: Done.

File: src/utils/CommonUtils.js (hunk 2 above)
Comment (BhagyashreeBhoyar) [reply src/utils/CommonUtils.js]: Done.

**Pull request merged**